diff --git a/drivers/clk/ti/clk-33xx.c b/drivers/clk/ti/clk-33xx.c
index 7dc30dd6c..728950606 100644
--- a/drivers/clk/ti/clk-33xx.c
+++ b/drivers/clk/ti/clk-33xx.c
@@ -197,7 +197,7 @@ static const struct omap_clkctrl_bit_data am3_debugss_bit_data[] __initconst = {
 };
 
 static const struct omap_clkctrl_reg_data am3_l3_aon_clkctrl_regs[] __initconst = {
-	{ AM3_L3_AON_DEBUGSS_CLKCTRL, am3_debugss_bit_data, CLKF_SW_SUP, "l3-aon-clkctrl:0000:24" },
+	{ AM3_L3_AON_DEBUGSS_CLKCTRL, am3_debugss_bit_data, CLKF_SW_SUP | CLKF_NO_IDLEST, "l3-aon-clkctrl:0000:24" },
 	{ 0 },
 };
 
